easybuild/easyconfigs/p/PyTorch/PyTorch-1.4.0-fosscuda-2019b-Python-3.7.4.eb

```diff
@@ -280,32 +280,23 @@ dependencies = [
     ('NCCL', '2.4.8'),
 ]
 
-download_dep_fail = True
-
-prebuildopts = 'PYTORCH_BUILD_VERSION=%(version)s PYTORCH_BUILD_NUMBER=1 VERBOSE=1 '
-prebuildopts += 'MAX_JOBS=%(parallel)s '
-prebuildopts += 'LDFLAGS="$LDFLAGS -ldl" '
-prebuildopts += 'USE_FFMPEG=ON USE_IBVERBS=1 USE_GFLAGS=ON USE_GLOG=ON '
-prebuildopts += 'CUDNN_LIB_DIR=$EBROOTCUDNN/lib64 CUDNN_INCLUDE_DIR=$EBROOTCUDNN/include '
-prebuildopts += 'USE_SYSTEM_NCCL=1 NCCL_INCLUDE_DIR=$EBROOTNCCL/include '
-prebuildopts += 'TORCH_CUDA_ARCH_LIST="3.5 3.7 5.2 6.0 6.1 7.0 7.2 7.5" '
-
-preinstallopts = prebuildopts
-
-runtest = 'export PYTHONPATH=%(builddir)s/%(namelower)s-%(version)s/' \
-          'build/lib.linux-%(arch)s-%(pyshortver)s:$PYTHONPATH '
-runtest += '&& export LD_LIBRARY_PATH=%(builddir)s/%(namelower)s-%(version)s/torch/lib64:$LD_LIBRARY_PATH '
-runtest += '&& python test/run_test.py --verbose '
-
-options = {'modulename': 'torch'}
-
-postinstallcmds = ['cp -a %(builddir)s/%(namelower)s-%(version)s/torch/lib*/* %(installdir)s/lib/']
-
-sanity_check_paths = {
-    'files': [],
-    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+excluded_tests = {
+    'POWER': [
+        # https://github.com/pytorch/pytorch/issues/41186
+        'autograd',
+        # Somehow only triggers on POWER, but is likely https://github.com/pytorch/pytorch/issues/33552
+        'distributed',
+        # https://github.com/pytorch/pytorch/issues/41400
+        'jit',
+        # https://github.com/pytorch/pytorch/issues/41469
+        'nn',
+        # Sporadically fails for unknown reason. Fixed in newer versions
+        'torch', 'distributions',
+    ],
 }
 
-sanity_pip_check = True
+# OpenMP threads must be limited as torch.matmul returns inprecise results for many threads (e.g. 176)
+runtest = 'cd test && OMP_NUM_THREADS=4 %(python)s run_test.py --verbose %(excluded_tests)s'
+tests = ['PyTorch-check-cpp-extension.py']
```

Comment:
the new easyblock tries to download this file and fails... where is it supposed to come from?

Reply:
Sorry, forgot to include it in this PR. Added to the commit.

Reply:
it seems easyblock needs to adjust this file's permissions so that it's executable

Reply:
Hm, strange. I checked it in as executable: `new file mode 100755` How did you test this? Probably a problem with EB?

Ccing @boegel

Reply:
I'm using `--from-pr`, which applies the PR patch on top of the target PR branch, so I guess `apply_patch` is not honouring the permissions (?)

Reply:
Yeah, it's possible file permissions get lost with `--from-pr`, maybe open an issue on that in the framework repo?

Why not run this with `python PyTorch-check-cpp-extension.py`, then permissions don't matter?

Reply:
Because EB expects that to be a filename, not a command. See `test_cases_step` of EasyBlock

Reply:
Permissions issue with `--from-pr` is fixed with https://github.com/easybuilders/easybuild-framework/pull/3414